vizro-core/docs/pages/user-guides/data.md

```diff
@@ -347,58 +347,59 @@ You cannot pass [nested parameters](parameters.md#nested-parameters) to dynamic
 
 ### Filters
 
-When a [filter](filters.md) depends on dynamic data and no `selector` is explicitly defined in the `vm.Filter` model, the available selector values update on page refresh to reflect the latest dynamic data. This is called a _dynamic filter_.
+When a [filter](filters.md) depends on dynamic data and no `selector` is explicitly defined in the `vm.Filter` model, the available selector values update either when the page refreshes or when a relevant [dynamic data parameter](#parametrize-data-loading) changes. This ensures the filter always reflects the latest data, which is called a _dynamic filter_.
```

Comment:
```suggestion
When a [filter](filters.md) depends on dynamic data and no `selector` is explicitly defined in the `vm.Filter` model, it is called a _dynamic filter_. A dynamic filter always reflects the latest data since the available selector values update either when the page refreshes or when a relevant [dynamic data parameter](#parametrize-data-loading) changes.
```